### PR TITLE
feat: add dynamicImportSettled utility function

### PR DESCRIPTION
**Pull request by sheremet-va**
Context #1328

### CODE REVIEW

**Comment by sheremet-va** [comment]
@cexbrayat what do you think about this?

File: packages/vitest/src/integrations/vi.ts [hunk 1]
```diff
@@ -234,6 +234,24 @@ class VitestUtils {
...
+   */
+  public async dynamicImportSettled() {
+    const state = getWorkerState()
+    const promises: Promise<unknown>[] = []
+    for (const mod of state.moduleCache.values()) {
+      if (!mod.exports && mod.promise)
+        promises.push(mod.promise)
+    }
+    await Promise.all(promises)
+    // wait untill the end of the loop, so `.then` on modules called,
```
Comment (cexbrayat) [inline packages/vitest/src/integrations/vi.ts:251]: nit: untill -> until

**Review by cexbrayat — APPROVED** [review]
That looks great 👍

File: packages/vitest/src/integrations/vi.ts (hunk 1 above)
Comment (antfu) [inline packages/vitest/src/integrations/vi.ts:251]:
```suggestion
    // wait until the end of the loop, so `.then` on modules called,
```

**Review by antfu — APPROVED** [review]

**Pull request merged**